wasatch/FeatureIdentificationDevice.py

```diff
@@ -209,8 +209,8 @@ def connect(self) -> SpectrometerResponse:
                 if "Resource busy" in str(exc):
                     log.warn("Hardware Failure in setConfiguration. Resource busy error. Attempting to reattach driver by reset.")
                     self.device_type.reset(dev)
-                    connect()
-                    return self.connected
+                    connect() # MZ: I don't see how this could work?  What is connect()?
```

Comment:
This is a typo it should be self.connect(). As the log statement says if we got a busy error then attempt to reset and connect again. The function would recurse and start over. I see some bad design in this though since there is no limit on the retries and if there is something seriously wrong then this could blow the stack.

Reply:
Okay, I made some changes around this.  Please review the new version and let me know what you think.